hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -112,7 +114,11 @@ public ClientMessage getClientMessage() {
 
     public ClientInvocationFuture invoke() {
         assert (clientMessage != null);
-        clientMessage.setCorrelationId(callIdSequence.next());
+        try {
+            clientMessage.setCorrelationId(callIdSequence.next());
+        } catch (TimeoutException e) {
+            throw new HazelcastOverloadException("Timed out trying to acquire another call ID.");
```

Comment:
would be good to have indication of amount of time that has been spend waiting. Can't we include the timeoutexception as cause?